Initial HomePage styling

Tijani-Dia opened the pull request:
Style all hardcoded components in the HomePage

jhancock532 commented on styles/_variables.scss:
```diff
@@ -0,0 +1,52 @@
+$font--primary: 'Inter';
```
**Suggestion** : Add a backup font for the font family as there are flashes of unstyled text when I resize the window occasionally

```suggestion
$font--primary: 'Inter', Helvetica, Arial;
```

jhancock532 commented on styles/globals.scss:
```diff
@@ -0,0 +1,17 @@
+body {
+    padding: 0;
+    margin: 0;
+    font-family: $font--primary;
+    background: $color--primary;
+    color: $color--white;
+}
+
+a {
+    color: #3beccd;
```
**Suggestion**: Consider colour variable refactor here

Tijani-Dia replied:
I wanted to do that, but I didn't know how to name it...🙂

jhancock532 replied:
See above colour conversation on theming

jhancock532 commented on pages/index.tsx:
```diff
@@ -1,23 +1,65 @@
 const Home: NextPage = () => {
     return (
         <div className={styles.container}>
             <Head>
                 <title>Are we headless yet? | Wagtail</title>
```
**Suggestion**: Maybe this is needed in a separate ticket, but we should add some social media metadata tags here ([CSS Tricks](https://css-tricks.com/essential-meta-tags-social-media/)) and maybe use a [favicon generator](https://realfavicongenerator.net/) to make all the favicon files?

jhancock532 replied:
If you're familiar with Codebase you could copy paste the above into a new ticket 🙂

jhancock532 reviewed (commented):
This is coming together well, some notes on SCSS best practices and thoughts on the next.js components, Link and Image. Using pixel values instead of rem for margins and padding would be nice, but this is a personal preference - Tailwind uses style rules that are fractions and multiples of rem, so we could follow that convention here as well.

Tijani-Dia commented on components/BottomCTA.tsx:
```diff
@@ -0,0 +1,28 @@
+    <div className={styles.container}>
+        <span>
+            <Image
+                src="/hear-icon.svg"
```
Alright!

Tijani-Dia commented on components/BottomCTA.tsx:
```diff
@@ -0,0 +1,28 @@
+
+export const BottomCTA = () => (
+    <div className={styles.container}>
+        <span>
```
Alright.

Tijani-Dia commented on next.config.js:
```diff
@@ -1,4 +1,8 @@
     reactStrictMode: true,
+    sassOptions: {
+        // allow all scss files access to these files
+        prependData: `@import "variables.scss"; @import "mixins.scss"; @import "fonts.scss"; `,
```
I haven't used it yet but if you think we should add it then we can go for it.

Tijani-Dia commented on pages/index.tsx:
```diff
@@ -1,23 +1,65 @@
+                            <span>headless</span> yet?
+                        </h1>
+                        <p className={styles.description}>
+                            👍 Yes! But there&apos;s plenty we could improve for
+                            Wagtail&apos;s developers
```
That's what I did initially but it was raising a lint issue. (See https://github.com/wagtail/areweheadlessyet/runs/5009282346?check_suite_focus=true)

jhancock532 replied:
Thank you, hopefully this isn't an issue in the future: if it is, you can disable specific linting rules [with this configuration option](https://nextjs.org/docs/basic-features/eslint#disabling-rules).

Tijani-Dia replied:
I've applied your suggestion and disabled the linting rule mentioned above.

Tijani-Dia replied:
>Additionally the Yes! should be a strong.

This is hardcoded right now but will be fetched from the API at some point. I will update the styling then

Tijani-Dia commented on public/hear-icon.svg:
```diff
@@ -0,0 +1,9 @@
+<svg width="32" height="33" viewBox="0 0 32 33" fill="none" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Alright, I've uploaded the PNG version.

Tijani-Dia commented on styles/BottomCTA.module.scss:
```diff
@@ -0,0 +1,54 @@
+    @include font-size(m);
+}
+
+.cta {
+    font-size: 30px;
+    line-height: 1.15;
+    font-weight: 900;
+    display: flex;
+    flex-direction: column;
+
+    @include media-query(large) {
+        display: block;
+        line-height: 1.5;
+    }
+}
```
Very nifty indeed.

Tijani-Dia commented on styles/Home.module.scss:
```diff
@@ -41,19 +65,20 @@
-.title,
-.description {
-    text-align: center;
+.title > span {
+    color: #3beccd;
 }
```
That makes sense. Hoewever, why do you suggest writing the list of color variables in `globals.scss` rather than in `_variables.scss`?

Tijani-Dia commented on components/BottomCTA.tsx:
```diff
@@ -0,0 +1,28 @@
+                alt="Hear the latest news"
+                width={32}
+                height={24}
+            />
```
Alright, I'll keep that in mind. Thanks!

Tijani-Dia commented on pages/index.tsx:
```diff
@@ -1,23 +1,65 @@
+                    </span>
+                    <div className={styles.hero_body}>
+                        <h1 className={styles.title}>
+                            Are we <br />
```
Yes, in this case, I found that it worked better than trying to do it in CSS.

Tijani-Dia commented on styles/Home.module.scss:
```diff
@@ -114,3 +139,44 @@
+    transform: rotate(159.81deg);
+    position: absolute;
+    right: -3rem;
+    bottom: -2rem;
```
What do you mean by 
>include a translation

?

Tijani-Dia commented on styles/_variables.scss:
```diff
@@ -0,0 +1,52 @@
+$color--white: #ffffff;
+
+$breakpoints: (
+    'small' '(max-width: 599px)',
+    'medium' '(min-width: 600px)',
```
That makes sense. I've removed the 'small' breakpoint.

Tijani-Dia commented on pages/index.tsx:
```diff
@@ -58,24 +100,33 @@ const Home: NextPage = () => {
+                <BottomCTA />
             </main>
 
             <footer className={styles.footer}>
```
I've made a separate component for the footer and (hopefully) fixed the issue on small mobile devices mentionned.

jhancock532 commented on components/Footer.tsx:
```diff
@@ -0,0 +1,30 @@
+import Image from 'next/image';
+import styles from '../styles/Footer.module.scss';
+
+export const Footer = () => (
+    <footer className={styles.footer}>
+        <div className={styles.footer__text}>
+            <div>
+                <span className={styles.copyright}>© Are we headless yet?</span>
+                <span>Updated 2022</span>
+            </div>
+            <span className={styles.delimiter}>-</span>
+            <div>
+                <a href="https://wagtail.org/" className={styles.link}>
+                    Wagtail
+                </a>{' '}
+                is an open source project
+            </div>
+        </div>
+        <div className={styles.logo}>
+            <Image
+                src="/images/headless-logo.svg"
+                alt="Headless Wagtail Logo"
+                width={127}
+                height={133}
+            />
+        </div>
+    </footer>
+);
```
**Praise**: Awesome, good to see a separate component for this 👍

jhancock532 commented on pages/index.tsx:
```diff
@@ -1,23 +1,68 @@
+                    </div>
+                </div>
 
                 <div className={styles.grid}>
```
If we're ready, we can delete all the boilerplate code here and in `Home.module.scss` that doesn't go into the final design. If it's still useful for testing and page layout feel free to keep it in

jhancock532 commented on styles/Footer.module.scss:
```diff
@@ -0,0 +1,58 @@
+        }
+    }
+
+    &__text {
+        display: flex;
+        flex-direction: column;
+        position: relative;
+        z-index: 1;
+        @include font-size(xxs);
```
```suggestion
    &__text {
        @include font-size(xxs);
        display: flex;
        flex-direction: column;
        position: relative;
        z-index: 1;
```

**Nitpick**: Nice, this is all looking good. Ben Morse taught me recently that `@include` rules which aren't media queries should go to the top of the class definition, I like this idea of laying out styles. We could add another ticket to Codebase for setting up a [Stylelint config](https://stylelint.io/) for this project which should catch things like this automatically.

jhancock532 commented on styles/Footer.module.scss:
```diff
@@ -0,0 +1,58 @@
+        right: 0;
+        top: 0;
+        z-index: 0;
+        opacity: 0.5;
```
This is a great touch & helps readability, awesome.

Tijani-Dia replied:
Thanks!

jhancock532 commented on styles/Home.module.scss:
```diff
@@ -10,22 +10,46 @@
+    }
 }
 
 .title a {
```
There are a few other class styles in this stylesheet that won't be in the final design - feel free to keep them as long as they are useful, we can delete them later so the stylesheet doesn't get cluttered

Tijani-Dia replied:
I'm keeping them right now so the page doesn't look empty but I plan to remove them as soon as we have real content.

jhancock532 commented on styles/Home.module.scss:
```diff
@@ -114,3 +100,26 @@
+        margin-bottom: 4rem;
+    }
+
+    &__body {
```
Nice 😄

jhancock532 commented on styles/_variables.scss:
```diff
@@ -0,0 +1,52 @@
+$font--primary: 'Inter';
+
+$color--primary: #183153;
+$color--white: #ffffff;
```
I've realised we've got a two color theme being used throughout this site, so it's worthwhile setting up the colours as :root variables now so we can change them dynamically later.

```css
:root {
    --color--white: #fff;
   --color--lagoon: #3beccd;
   --color--purple: #183153;
   ...
   --color--text: var(--color--white);
   --color--background: var(--color--purple);
   --color--accent: var(--color--lagoon);
}
```

Throughout the source code, everything would be styled with the colors of the generic root variable names, so these colours can update dynamically later using [React snippets like this](https://stackoverflow.com/questions/45226712/how-to-change-css-root-variable-in-react). However, I haven't researched colour theming in React before, you might find a better solution (this solution is based off how `wagtail-torchbox` works)

Tijani-Dia replied:
I've added it under the `:root` ruleset. Thanks!

jhancock532 commented on styles/_variables.scss:
```diff
@@ -0,0 +1,52 @@
+);
+
+// Font sizes
+$base-font-size: 18px;
```
Alternatively, just delete `$base-font-size` as it's not being used and add `html { font-size: 16px; }` to ensure browser consistency with rem styling. (Apologies, it's not `body` that changes REM)

jhancock532 reviewed (commented):
Thanks for making the changes, I've clarified my thoughts on colour styling for this site as the designs include colour theming. It would be nice to plan this out in advance to avoid refactoring later.

thibaudcolas commented:
@Tijani-Dia additional review comments based on testing the page in Chrome & Safari:

- Please take a moment in your PR description to describe the changes – which tickets your work corresponds to, anything that’s intended to still be a work-in-progress vs. finished, how you’ve tested this, how you expect reviewers to test the changes
- The `<html>` tag is missing a `lang="en"`
- The styles of the "highlight" divider’s linecaps don’t look correct to me
- The hear emoji appears squashed for me. I think we should probably switch back from the PNG to the text symbol instead, since we’re doing this for the thumbs up.
- On smaller viewport widths, the logo appears underneath the headline text.

Tijani-Dia commented on components/BottomCTA.tsx:
```diff
@@ -0,0 +1,22 @@
+import Image from 'next/image';
+import styles from '../styles/BottomCTA.module.scss';
```
Alright.

Tijani-Dia commented on components/Footer.tsx:
```diff
@@ -0,0 +1,30 @@
+        <div className={styles.footer__text}>
+            <div>
+                <span className={styles.copyright}>© Are we headless yet?</span>
+                <span>Updated 2022</span>
```
That's right, I will do it as part of the API fetching work.

Tijani-Dia commented on pages/index.tsx:
```diff
@@ -1,23 +1,68 @@
+                <link rel="icon" href="/images/headless-logo.svg" />
+                <link
+                    rel="preload"
+                    href="/fonts/inter/inter-medium.woff2"
```
I've removed it.

Tijani-Dia commented on public/images/slack-icon.svg:
```diff
@@ -0,0 +1 @@
+<svg width="32" height="33" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M6.72 20.588c0 1.837-1.496 3.322-3.347 3.322S.025 22.425.025 20.588c0-1.838 1.497-3.322 3.348-3.322H6.72v3.322ZM8.394 20.588c0-1.838 1.496-3.322 3.347-3.322 1.852 0 3.348 1.485 3.348 3.322v8.305c0 1.837-1.496 3.322-3.348 3.322-1.85 0-3.347-1.485-3.347-3.322v-8.305ZM11.741 7.249c-1.85 0-3.347-1.485-3.347-3.322S9.89 ...
```
>Is it expected that the width and height are different?

I downloaded it as is and I didn't change it.

I've addressed the remaining points.

Tijani-Dia commented on styles/_fonts.scss:
```diff
@@ -0,0 +1,31 @@
+    src: url('/fonts/inter/inter-regular.woff2') format('woff2');
+    font-weight: normal;
+    font-style: normal;
+    font-display: swap;
```
I've set it to `block` for now. Let me know if you think that we should use `fallback`.

Tijani-Dia commented on styles/_mixins.scss:
```diff
@@ -0,0 +1,40 @@
+*/
+
+@mixin font-size($size) {
+    @if $size == null {
+        @warn 'Font size ‘#{$keyword}’ does not exist';
+    } @else {
```
To be honest, I copied it from [here](https://github.com/torchbox/careers/blob/07ee80a8922a7990c1add2f5ac6910a826c579dc/styles/_mixins.scss#L55-L72) without fully understanding it.

Tijani-Dia commented on pages/index.tsx:
```diff
@@ -1,23 +1,68 @@
             <Head>
                 <title>Are we headless yet? | Wagtail</title>
-                <link rel="icon" href="/favicon.ico" />
+                <link rel="icon" href="/images/headless-logo.svg" />
```
I used a [favicon generator](https://realfavicongenerator.net/) as suggested by @jhancock532 to generate the `.ico` file (and replace the current SVG one).

Tijani-Dia commented on styles/_variables.scss:
```diff
@@ -0,0 +1,54 @@
+    xs: 14px,
+    xxs: 13px,
+    xxxs: 12px,
+);
```
It's done.

Tijani-Dia commented on components/TopCTA.tsx:
```diff
@@ -0,0 +1,20 @@
+import Image from 'next/image';
+import styles from '../styles/TopCTA.module.scss';
+
+export const TopCTA = () => (
```
I renamed it so.

Tijani-Dia commented on components/BottomCTA.tsx:
```diff
@@ -0,0 +1,22 @@
+import Image from 'next/image';
+import styles from '../styles/BottomCTA.module.scss';
+
+export const BottomCTA = () => (
```
I was not very inspired for naming these componenents. I've renamed them following @jhancock532's suggestion.

Tijani-Dia commented:
Thanks @thibaudcolas and @jhancock532 for your thorough reviews!

@thibaudcolas 
> The <html> tag is missing a lang="en"

I addressed this following what has been described [here](https://melvingeorge.me/blog/set-html-lang-attribute-in-nextjs).


> The styles of the "highlight" divider’s linecaps don’t look correct to me

What do you mean by `highlight divider’s linecaps`?

> The hear emoji appears squashed for me. I think we should probably switch back from the PNG to the text symbol instead, since we’re doing this for the thumbs up.

Addressed.

> On smaller viewport widths, the logo appears underneath the headline text.

Can you share a screenshot of this please?

> Please take a moment in your PR description to describe the changes – which tickets your work corresponds to, anything that’s intended to still be a work-in-progress vs. finished, how you’ve tested this, how you expect reviewers to test the changes

I didn't want to link to the corresponding ticket since people who don't have access to the project (in Codebase) won't be able to access it. I agree that I could describe better the changes. To test it, the main criteria is that the design matches the one in Figma. This ticket only includes hardcoded components.

thibaudcolas commented:
@Tijani-Dia sorry, I should have shared screenshots right away 🤦  here we are.

I believe this sound be round on the ends:

![Screenshot 2022-02-03 at 16 09 30](https://user-images.githubusercontent.com/877585/152381483-17e404d5-3db1-405e-9601-eac53101a2b1.png)

And here is the logo overlapping:

![Screenshot 2022-02-03 at 16 09 20](https://user-images.githubusercontent.com/877585/152381410-85e2ecca-740c-4504-b2e3-16d09c9d2149.png)

Tijani-Dia commented:
@thibaudcolas I've addressed the styling issues mentioned in the previous comment.

thibaudcolas commented on styles/_mixins.scss:
```diff
@@ -0,0 +1,29 @@
+*/
+
+@mixin font-size($size) {
+    @if $size == null {
+        @warn 'Font size ‘#{$keyword}’ does not exist';
+    } @else {
+        font-size: map-get($font-sizes, $size);
+    }
```
```suggestion
    font-size: map-get($font-sizes, $size);
```

thibaudcolas reviewed (approved):
Looking good! I’ve suggested one additional change based on my previous review. Additionally I really think the logo should have been done with inline SVG so we can change its color based on the theme more easily – but since it’s done already let’s wait for dark mode to be added before we change that.

Tijani-Dia commented:
Thanks @thibaudcolas. I've done the logo with inline SVG in the following PR.

thibaudcolas commented:
Ah nice :)
